core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackService.java

```diff
@@ -536,8 +549,15 @@ public Stack create(Stack stack, String platformString, StatedImage imgFromCatal
         }, LOGGER, "Security config save took {} ms for stack {}", stackName);
         savedStack.setSecurityConfig(securityConfig);
 
-        // JSA todo populate CCM parameters
         CcmParameters ccmParameters = null;
+        if ((ccmParameterSupplier != null) && Boolean.TRUE.equals(stack.getUseCcm())) {
+            int gatewayPort = Optional.ofNullable(stack.getGatewayPort()).orElse(ServiceFamilies.GATEWAY.getDefaultPort());
+            Map<KnownServiceIdentifier, Integer> tunneledServicePorts = Collections.singletonMap(KnownServiceIdentifier.GATEWAY, gatewayPort);
+            // JSA TODO Use stack ID or something else instead?
+            String keyId = StringUtils.right(stack.getResourceCrn(), CCM_KEY_ID_LENGTH);
```

Comment:
The KeyId is what we use to look up the tunnel when we want to use it, right? shouldn't we save it as part of the stack model so we can retrieve it later?